File: restoGhid-backend/src/main/java/com/example/licentaRestaurante/entities/Restaurant.java
package com.example.licentaRestaurante.entities;

import com.fasterxml.jackson.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "restaurant")
@AllArgsConstructor
@RequiredArgsConstructor
@JsonIgnoreProperties(value = {"hibernateLazyInitializer", "handler"}, ignoreUnknown = true)
public class Restaurant {
    @Id
    @Column(name = "location_id", nullable = false)
    @JsonProperty("location_id")
    String locationId;

    @JsonProperty("name")
    String name;

    @JsonProperty("latitude")
    String latitude;

    @JsonProperty("longitude")
    String longitude;

    @JsonProperty("num_reviews")
    String num_reviews;

    @JsonProperty("photo")
    @JsonInclude()
    @Transient
    Photo photo;

    @JsonProperty("ranking_geo")
    String ranking_geo;

    @JsonProperty("rating")
    String rating;

    @JsonProperty("is_closed")
    boolean isClosed;

    @JsonProperty("price")
    String price;

    @JsonProperty("description")
    String description;

    @JsonProperty("category")
    @JsonInclude()
    @Transient
    Category category;

    @Column(name = "category_key")
    private String categoryKey;

    @JsonProperty("subcategory")
    @JsonInclude()
    @Transient
    ArrayList<Subcategory> subcategory;

    @Column(name = "subcategory_key")
    private String subcategoryKey;

    @JsonProperty("phone")
    String phone;

    @JsonProperty("website")
    String website;

    @JsonProperty("email")
    String email;

    @JsonProperty("address")
    String address;

    @JsonProperty("cluster")
    Integer cluster;

    @Column(name = "manager_username")
    String managerUsername;

    @Column(name = "total_seats")
    @JsonProperty("total_seats")
    Integer totalSeats;

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "restaurant_cuisine",
            joinColumns = {@JoinColumn(name = "restaurant_id")},
            inverseJoinColumns = {@JoinColumn(name = "cuisine_key")})
    @JsonProperty("cuisine")
    private Set<Cuisine> cuisines = new HashSet<>();

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "restaurant_dietary",
            joinColumns = {@JoinColumn(name = "restaurant_id")},
            inverseJoinColumns = {@JoinColumn(name = "dietary_key")})
    @JsonProperty("dietary_restrictions")
    private Set<DietaryRestriction> dietaryRestrictions = new HashSet<>();

    @OneToMany(mappedBy = "restaurant")
    private List<Review> reviews = new ArrayList<>();

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL)
    @JsonManagedReference("restaurant-booking")
    private List<Booking> bookings;

    // add booking
    public void addBooking(Booking booking) {
        if (bookings == null) {
            bookings = new ArrayList<>();
        }
        bookings.add(booking);
        booking.setRestaurant(this);
    }

    // remove booking
    public void removeBooking(Booking booking) {
        if (bookings != null) {
            bookings.remove(booking);
            booking.setRestaurant(null);
        }
    }
}

File: restoghid-android/app/src/main/java/com/example/licenta/models/Restaurant.kt
package com.example.licenta.models

import com.google.gson.annotations.SerializedName

data class Restaurant(
    @SerializedName("categoryKey")
    val categoryKey: String,
    @SerializedName("subcategoryKey")
    val subcategoryKey: String,
    @SerializedName("location_id")
    val locationId: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("latitude")
    val latitude: String,
    @SerializedName("longitude")
    val longitude: String,
    @SerializedName("manager_username")
    val managerUsername: String,
    @SerializedName("num_reviews")
    val numReviews: String,
    @SerializedName("ranking_geo")
    val rankingGeo: String,
    @SerializedName("rating")
    val rating: String,
    @SerializedName("is_closed")
    val isClosed: Boolean,
    @SerializedName("price")
    val price: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("phone")
    val phone: String,
    @SerializedName("website")
    val website: String,
    @SerializedName("email")
    val email: String,
    @SerializedName("address")
    val address: String,
    @SerializedName("cluster")
    val cluster: Int,
    @SerializedName("total_seats")
    val totalSeats: Int,
    @SerializedName("cuisine")
    val cuisine: MutableList<Cuisine>,
    @SerializedName("dietary_restrictions")
    val dietaryRestrictions: MutableList<DietaryRestrictions>,
//    @SerializedName("reviews")
//    val reviews: MutableList<Review>,
    @SerializedName("bookings")
    val bookings: MutableList<Booking>
)

data class Cuisine(
    @SerializedName("key")
    val key: String,
    @SerializedName("name")
    val name: String
)

data class DietaryRestrictions(
    @SerializedName("key")
    val key: String,

    @SerializedName("name")
    val name: String
)